js/Integer.js

```diff
@@ -0,0 +1,64 @@
+const InvalidInput = require("./Errors")
+
+class Integer {
+    static plus(a, b) {
+        Integer.#validate(a, b)
+
+        let carry = 0
+        while (b !== 0) {
+            carry = a & b
+            a = a ^ b
+            b = carry << 1
+        }
+
+        return a
+    }
+
+    static product(a, b) {
+        Integer.#validate(a, b)
+
+        let signedBit = 1 << 31
+        let isNegativeResult = false
+        if (a & signedBit) {
+            a = Integer.negate(a)
+            isNegativeResult = !isNegativeResult
+        }
+        if (b & signedBit) {
+            b = Integer.negate(b)
+            isNegativeResult = !isNegativeResult
+        }
+
+        let result = 0
+        while (a !== 0) {
+            if ((a & 1) === 1) {
+                result += b
+            }
+            a = a >>> 1
+            b = b << 1
+        }
+
+        if (isNegativeResult) {
+            return Integer.negate(result)
+        } else {
+            return result
+        }
+    }
+
+    static negate(integer) {
+        let bitflipSummand = -1 >> 32
+        return ((integer ^ bitflipSummand) + 1)
+    }
```

Comment:
1) Illegal use of `-` and `+` operators (lines 48 and 49).
2) `negate` does two things: flip bits and add one. It would be cleaner to use a sub-method like so:
    ```
    static negate(integer) {
        return Integer.plus(Integer.#flipBits(integer), 1)
    }

    static #flipBits(integer) {
        // ...
    }
    ```
I'll leave it to you to find out how to implement `Integer.#flipBits` without using `-`. :-)